### ISSUE TITLE
[Bug]: Error during "clearing" a transaction with no Payee

### ISSUE THREAD

**Issue by KuroSetsuna29** [actualbudget/actual#1896]
### Verified issue does not already exist?

- [X] I have searched and found no existing issue

### What happened?

When you try to clear a transaction that has no Payee set, I get the red "Something internally went wrong." toast message.

This also seems to break importing records, the record doesn't show up on the other accounts that have the Payee automatically set up as a transfer, unless I manually edit the record first. I think it fails processing an earlier transaction without a Payee and errors out without processing the rest of the records properly.

### What error did you receive?

From the browser console:
```
[Exception] TypeError: Cannot destructure property 'transfer_acct' of '(intermediate value)' as it is null.
```

### Where are you hosting Actual?

Docker

### What browsers are you seeing the problem on?

Chrome

### Operating System

Windows 10

**Comment by shaankhosla** [comment]
I'm not able to replicate this, could you provide a screenshot or a short screen recording with this bug on https://demo.actualbudget.org?

**Comment by KuroSetsuna29** [comment]
Hi, I was unable to replicate the issue on the demo site.

After debugging on my local instance I found the issue was that during import it was attempting to set the Payee to an non-existent Payee in the DB. It turned out I found some rules that was trying to set the Payee to a `(deleted)` payee. I am not sure how that payee was deleted, I could have possibly deleted that payee before and the Rule was just never updated. 

It looks like after fixing the Rules that displayed `(deleted)` payee, all new imported transactions are fine, and after updating all the existing transactions by updating with a random Payee and then clearing it again seems to also fix the issue with the existing transactions.

Also, it looks like I was running version v23.10.0, and after upgrading to v23.11.0 I was unable to reproduce the error via the same steps. (I still see rule with `(deleted)` payee but no more errors while trying to import transactions with the matching rule)

Thanks for looking into this, sorry for the trouble. Since this seems to be fixed in new version I am closing this issue.
